io/eolearn/io/processing_api.py

```diff
@@ -98,64 +65,118 @@ def execute(self, eopatch=None, bbox=None, time_interval=None):
         elif self.resolution is not None:
             size_x, size_y = bbox_to_dimensions(eopatch.bbox, self.resolution)
 
-        eopatch.timestamp = self.get_dates(eopatch.bbox, time_interval)
-
-        payloads = (self._request_payload(date, eopatch.bbox, size_x, size_y) for date in eopatch.timestamp)
+        timestamp = self._get_timestamp(time_interval, bbox) if time_interval else None
 
-        request_args = dict(
-            url=self.config.get_sh_processing_api_url(),
-            headers={"accept": "application/tar", 'content-type': 'application/json'},
-            data_folder=self.cache_folder,
-            hash_save=bool(self.cache_folder),
-            request_type='POST',
-            data_type=MimeType.TAR
-        )
-        requests = [DownloadRequest(post_values=payload, **request_args) for payload in payloads]
+        payloads = self._build_payloads(bbox, size_x, size_y, timestamp)
+        requests = [DownloadRequest(post_values=payload, **self.request_args) for payload in payloads]
 
         LOGGER.debug('Downloading %d requests of type %s', len(requests), str(self.data_source))
-        LOGGER.debug('Downloading bands: [%s]', ', '.join(self.all_bands))
         client = SentinelHubDownloadClient(config=self.config)
         images = client.download(requests, max_threads=self.max_threads)
         LOGGER.debug('Downloads complete')
 
-        images = ((img['default.tif'], img['userdata.json']) for img in images)
-        images = [(img, meta.get('norm_factor', 0)) for img, meta in images]
+        temporal_dim = len(timestamp) if timestamp else 1
```

Comment:
This one is important: 
As far as I can see `timestamp` variable is never passed to `eopatch.timestamp`.

Besides fixing this we should also improve unit tests so that they would test such a bug.